airflow/providers/apache/cassandra/provider.yaml

```
@@ -33,10 +35,15 @@ integrations:
 sensors:
   - integration-name: Apache Cassandra
     python-modules:
-      - airflow.providers.apache.cassandra.sensors.record
-      - airflow.providers.apache.cassandra.sensors.table
+      - .sensors.record
+      - .sensors.table
 
 hooks:
   - integration-name: Apache Cassandra
     python-modules:
-      - airflow.providers.apache.cassandra.hooks.cassandra
+      - .hooks.cassandra
+
+connections:
+  - connection-type: cassandra
+    hook-class: .hooks.cassandra.CassandraHook
+    connection-id-parameter-name: cassandra_conn_id
```

I'd rather we defined these as properties in code than in a yaml file
```suggestion
connections:
  - hook-class: .hooks.cassandra.CassandraHook
```

And then:

```python
class CassandraHook(BaseHook):
   CONNECTION_TYPE="cassandra"
   CONNECTION_ID_PARAM_NAME="cassanrda_conn_id"
```

---

There are many other things that are not yet defined here (other PRs are addressing it) - extra links and connection in forms in javascript. If you look at those, you will understand that those cannot be as easily incorporated in the classes. 

I prefer to keep "provider delivered functionality" in one place rather than some of it in .yaml, some of this in some connection and some of this in other classes.

Also, our proposal is far from complete - it does not actually solve the problem of discovering those Hooks (unless you have not posted the entry_points mechanism, you will have to somehow at least keep the list of hooks in each provider (there are at least 4 in the google provider). That would mean that you would have to not only keep the list of hooks but also remember to update the  "extra" information in the hooks - this will have to be (by its nature) split between two places. 

What was your idea here? where did you want to keep the list of connections to load?